src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -1414,14 +1414,7 @@ bool ICollection.IsSynchronized
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         object ICollection.SyncRoot
         {
-            get
-            {
-                // As a struct, this instance will be boxed in order to call this property getter.
-                // We will return the boxed instance. If the caller boxes and reboxes repeatedly,
-                // they'll get a different SyncRoot each time.
-                // No real alternative, but we don't anticipate this will be much problem anyway.
-                return this;
-            }
+            get { throw new NotSupportedException(); }
```

Comment:
:exclamation: Can we not use `SR.ConcurrentCollection_SyncRoot_NotSupported` (either directly, or by copying that resource to this assembly and naming it `Immutable[Collection|Array]_SyncRoot_NotSupported`?